sources/Tools/Visualiser.py

```diff
@@ -126,26 +125,51 @@ def _process_bar_data(raw_data, head):
             "left": [(stat[index] == "Left").sum() for index in range(4)],
         }
 
-    def _histogram_visualizer(self, head):
-        logging.warning(
-            f"{inspect.currentframe().f_code.co_name}:Need to be refactor - No hard coded data please"
-        )
-        hands = self._process_bar_data(self.raw_data, head)
-        houses = set(self.raw_data.loc[:, "Hogwarts House"])
-        x = np.arange(len(houses))
-        width = 0.35
-        fig, ax = plt.subplots()
-        # print(hands)
-        rects1 = ax.bar(x - width / 2, hands["right"], width, label="Right")
-        rects2 = ax.bar(x + width / 2, hands["left"], width, label="Left")
-        ax.set_xticks(x)
-        ax.set_xticklabels(houses)
-        ax.legend()
-        self._auto_label(ax, rects1)
-        self._auto_label(ax, rects2)
-        fig.tight_layout()
+    @staticmethod
+    def _pick_by_elements(raw_data, filters, column_filter, feature):
+        return [raw_data.loc[raw_data[column_filter] == element, feature] for element in filters]

```

Comment:
check if numpy.where could be useful

Reply:
i've already try it with pandas.query() but this is the most handle solution since we need to pick element from filter IN feature and we can't cut the data before to do it since we need to compare and check all theses terms at the same time.